### PR TITLE
PT-1444 Deduplicate PAs based on EventIDs

### PR DESCRIPTION
**Pull request by franck-fourel**
EventID is a new field that identify uniquely some events. 

If the database contains several PA events with the same EventId, we should display only the most recent one. The other ones will be ignored.

### CODE REVIEW

File: js/tidelinedata.js [hunk 1]
```diff
@@ -229,6 +229,16 @@ function TidelineData(data, opts) {
     }
   };
 
+  this.deduplicatePhysicalActivities = function (data = []) {
+    var physicalActivity = _.groupBy(_.filter( data,  {type: 'physicalActivity'}), 'eventId');
```
Comment (yvon-dblg) [inline js/tidelinedata.js]:
```suggestion
    var physicalActivity = _.groupBy(_.filter( data, {type: 'physicalActivity'}), 'eventId');
```

File: js/tidelinedata.js [hunk 2]
```diff
@@ -229,6 +229,16 @@ function TidelineData(data, opts) {
...
+    _.forEach(physicalActivity, function(value, key) {
+      physicalActivity[key] = _.orderBy(value, ['deviceTime'],['desc']);
+    })
+    this.physicalActivities = _.map(physicalActivity, function(value) {
+      return value[0];
+    });
```
Comment (yvon-dblg) [inline js/tidelinedata.js]:
```suggestion
    this.physicalActivities = _.map(physicalActivity, (value) => value[0]);
```

File: js/tidelinedata.js [hunk 3]
```diff
@@ -229,6 +229,16 @@ function TidelineData(data, opts) {
     }
   };
 
+  this.deduplicatePhysicalActivities = function (data = []) {
+    var physicalActivity = _.groupBy(_.filter( data,  {type: 'physicalActivity'}), 'eventId');
+    _.forEach(physicalActivity, function(value, key) {
```
Comment (yvon-dblg) [inline js/tidelinedata.js]:
```suggestion
    _.forEach(physicalActivity, (value, key) => {
```

File: js/tidelinedata.js [hunk 4]
```diff
@@ -229,6 +229,16 @@ function TidelineData(data, opts) {
...
+      physicalActivity[key] = _.orderBy(value, ['deviceTime'],['desc']);
```
Comment (yvon-dblg) [inline js/tidelinedata.js]:
```suggestion
      physicalActivity[key] = _.orderBy(value, ['deviceTime'], ['desc']);
```

File: js/plot/physicalActivity.js [hunk 5]
```diff
@@ -35,7 +35,8 @@ module.exports = function(pool, opts) {
   return function(selection) {
     opts.xScale = pool.xScale().copy();
 
-    selection.each(function(currentData) {
+    selection.each(function() {
+      var currentData = opts.data;
```
Comment (yvon-dblg) [inline js/plot/physicalActivity.js:39]: Why this change was needed ?

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
It works 👍 
Few minor changes requested.

File: js/plot/physicalActivity.js (hunk 5 above)
Comment (franck-fourel) [reply js/plot/physicalActivity.js:39]: it's required to use the data passed to the component through opts, see line 262 of `chartdailyFactory.js`. 
Otherwise it would continue to use the default set of data based on the data type, physicalActivity.

File: package.json [hunk 6]
```diff
@@ -16,7 +16,7 @@
     "prebuild": "npm run clean",
     "build": "NODE_ENV=production webpack --optimize-minimize",
     "build-dev": "NODE_ENV=development webpack --progress",
-    "pretest": "npm run lint",
+    "apretest": "npm run lint",
```
Comment (yvon-dblg) [inline package.json]:
```suggestion
    "pretest": "npm run lint",
```
good for dev, not for commit ;-)

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
Missing to remove one dev stuff
Or you may remove this target, and add it to the .travis.yml

**Review by yvon-dblg — APPROVED** [review]
LGTM
